src/System.IO.FileSystem/tests/Performance/Perf.FileStream.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Xunit.Performance;
+using Xunit;
+
+namespace System.IO.Tests
+{
+    public class Perf_FileStream : FileSystemTest
+    {
+        private static byte[] CreateBytesToWrite(int size)
+        {
+            PerfUtils utils = new PerfUtils();
+            return Text.Encoding.UTF8.GetBytes(utils.CreateString(size));
+        }
+
+        private string CreateFile(int size)
+        {
+            PerfUtils utils = new PerfUtils();
+            string filePath = GetTestFilePath();
+            byte[] bytes = Text.Encoding.UTF8.GetBytes(utils.CreateString(size));
+            File.WriteAllBytes(filePath, bytes);
+            return filePath;
+        }
+
+        [Benchmark]
+        [InlineData(2000000)]
+        [InlineData(100000000)]
+        public void Write(int size)
+        {
+            string filePath = GetTestFilePath();
+            byte[] bytes = CreateBytesToWrite(size);
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (FileStream writer = File.Create(filePath))
+                using (iteration.StartMeasurement())
+                {
+                        writer.Write(bytes, 0, size);
+                }
+                File.Delete(filePath);
+            }
+        }
+
+        [Benchmark]
+        [InlineData(2000000)]
+        [InlineData(100000000)]
+        public void Read(int size)
+        {
+            string filePath = CreateFile(size);
+            byte[] bytes = new byte[size];
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (FileStream reader = File.OpenRead(filePath))
+                using (iteration.StartMeasurement())
+                {
+                    reader.Read(bytes, 0, size);
+                }
+            }
+            File.Delete(filePath);
+        }
+    }
+}
```

Comment:
Adding them now. How is this structure?

```
    if (useAsync)
    {
        // perform a number of writes of size bufferSize asynchronously, then waits for them all to complete
        for (int i = 0; i < innerIterations; i++)
            asyncCalls[i] = reader.ReadAsync(bytes, 0, bufferSize);
        Task.WaitAll(asyncCalls);
    }
    else
    {
        for (int i = 0; i < innerIterations; i++)
            reader.Read(bytes, 0, bufferSize);
    }
```
The results show the async versions slower in most cases, so I'm wondering if that structure is adding too much overhead in the WaitAll or something. I could await on every indiividual async call, but that just be the same test as the synchronous reads.